src/raftstore/store/peer.rs

```diff
@@ -187,6 +191,10 @@ pub struct Peer {
     pub tag: String,
 
     pub last_compacted_idx: u64,
+
+    become_candidate_time: Option<Timespec>,
```

Comment:
I don't think this is necessary. When a peer becomes leader, all lease read will fallback to quorum read.